### PR TITLE
feat: Update Dev Workspace Operator v0.20.0

### PR DESCRIPTION
**Pull request by tolusha**
### What does this PR do?
Update Dev Workspace Operator v0.20.0

### Screenshot/screencast of this PR

N/A

### What issues does this PR fix or reference?

Required step for https://github.com/eclipse/che/issues/22012

### How to test this PR?

N/A

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [Development resource are up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#update-development-resources)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: go.mod
[the remark this answers is not in the document]
Comment (amisevsk) [reply go.mod:1]: Let me know if I can help -- between v0.19 and v0.20, DWO had to update the kube API dependency which bumped a lot of versions.

[the remark this answers is not in the document]
Comment (amisevsk) [reply go.mod:1]: DWO issue/PR for reference: https://github.com/devfile/devworkspace-operator/issues/1059, https://github.com/devfile/devworkspace-operator/pull/1060

**Comment by AObuchow** [comment]
@tolusha here's the [PR](https://github.com/eclipse-che/che-operator/pull/1661) to add the changes to the `DEPENDENCIES.md` to the `update-dw20` branch.

**Comment by tolusha** [comment]
I can start a workspace with manual tests.

https://github.com/eclipse/che/issues/22153 for `ci/prow/v12-devworkspace-happy-path`

**Pull request merged**